TM1py/Services/CubeService.py

```diff
@@ -84,11 +84,19 @@ def get_control_cubes(self, **kwargs) -> List[Cube]:
         cubes = [Cube.from_dict(cube_as_dict=cube) for cube in response.json()['value']]
         return cubes
 
-    def get_number_of_cubes(self, **kwargs) -> int:
-        url = format_url(
-            "/api/v1/Cubes/$count")
-        response = self._rest.GET(url, **kwargs)
-        return int(response.text)
+    def get_number_of_cubes(self, skip_control_cubes: bool = False, **kwargs) -> int:
+        """ Ask TM1 Server for count of cubes
+
+        :skip_control_cubes: bool, True will exclude control cubes from count
+        :return: int, count
+        """ 
+        if skip_control_cubes:
+            response = len(self.get_all_names(skip_control_cubes=True, **kwargs))
```

Comment:
I imagine it would be more efficient if you used `ModelCubes()/$count` here

Reply:
It certainly would be more efficient except `ModelCubes()/$count`  or `ModelCubes/$count` don't work like `Cubes/$count` does unfortunately. I got lazy and leveraged len on the response of the get_all_names function. I've got an alternate method that grabs the @odata.count value from a response that I'll update today.